sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClient.java

```diff
@@ -0,0 +1,174 @@
+     *
+     * @param partitionId The unique identifier of a partition associated with the Event Hub.
+     * @return The information for the requested partition under the Event Hub this client is associated with.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public PartitionProperties getPartitionProperties(String partitionId) {
+        return client.getPartitionProperties(partitionId).block(retry.tryTimeout());
+    }
+
+    /**
+     * Creates an Event Hub producer responsible for transmitting {@link EventData} to the Event Hub, grouped together
+     * in batches. Event data is automatically routed to an available partition.
+     *
+     * @return A new {@link EventHubProducer}.
+     */
+    public EventHubProducer createProducer() {
+        return createProducer(defaultProducerOptions);
+    }
+
+    /**
+     * Creates an Event Hub producer responsible for transmitting {@link EventData} to the Event Hub, grouped together
+     * in batches. If {@link EventHubProducerOptions#partitionId() options.partitionId()} is not {@code null}, the
+     * events are routed to that specific partition. Otherwise, events are automatically routed to an available
+     * partition.
+     *
+     * @param options The set of options to apply when creating the producer.
+     * @return A new {@link EventHubProducer}.
+     * @throws NullPointerException if {@code options} is {@code null}.
+     */
+    public EventHubProducer createProducer(EventHubProducerOptions options) {
+        final EventHubAsyncProducer producer = client.createProducer();
+
+        return new EventHubProducer(producer, options.retry().tryTimeout());
+    }
+
+    /**
+     * Creates an Event Hub consumer responsible for reading {@link EventData} from a specific Event Hub partition, as a
+     * member of the specified consumer group, and begins reading events from the {@code eventPosition}.
+     *
+     * The consumer created is non-exclusive, allowing multiple consumers from the same consumer group to be actively
+     * reading events from the partition. These non-exclusive consumers are sometimes referred to as "Non-epoch
+     * Consumers".
+     *
+     * @param consumerGroup The name of the consumer group this consumer is associated with. Events are read in
+     *         the context of this group. The name of the consumer group that is created by default is {@link
+     *         EventHubAsyncClient#DEFAULT_CONSUMER_GROUP_NAME "$Default"}.
+     * @param partitionId The identifier of the Event Hub partition.
+     * @param eventPosition The position within the partition where the consumer should begin reading events.
+     * @return A new {@link EventHubConsumer} that receives events from the partition at the given position.
+     * @throws NullPointerException If {@code eventPosition}, or {@code options} is {@code null}.
+     * @throws IllegalArgumentException If {@code consumerGroup} or {@code partitionId} is {@code null} or an
```

Comment:
We should throw one type of exception when any input param is null/empty instead of throwing `NullPointerException`  for some and `IllegalArgumentException` for others. Thoughts?

Reply:
Yeah. I'll split the checks, wanted to use use the ImplUtils.isnullorEmpty and throw, but I'll differentiate based on the java spec for errors:
https://azure.github.io/azure-sdk/java_implementation.html